### PR TITLE
Restructure folder structure

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

The current folder structure isn't scalable and we should rework it.

This PR proposes that we create feature groups in the root of the `bitwarden` package. One such group is auth.

```
auth
  comands
  request
  response
  api
    request
    response
```

If needed we can group multiple features under a broader name, potentially the team name which allows for easy code ownership assignment.

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)

### CODE REVIEW

File: crates/bitwarden/src/auth/mod.rs [hunk 1]
```diff
@@ -1,2 +1,4 @@
+pub mod api;
```
Comment (Hinton) [inline crates/bitwarden/src/auth/mod.rs:1]: This should probably not be public, however we use api response in captcha response. Might be worth figuring that out.

Comment (dani-garcia) [reply crates/bitwarden/src/auth/mod.rs:1]: Would using `pub(super) mod api;` help here? That way it'll only be accesible to the auth module.

**Comment by dani-garcia** [comment]
Looks good overall to me, is the separation between `auth/[request,response]` and `auth/api/[request,response]` simply based on "this is the part that deals with client<->sdk" and "this is the part that deals with sdk<->server"?

Should we do something with the sdk module, that also has a lot of unrelated stuff inside?

**Comment by Hinton** [comment]
Pretty much. Sdk is now gone.

**Review by MGibson1 — APPROVED** [review]

**Review by MGibson1 — APPROVED** [review]

**Pull request merged**